### ISSUE TITLE
mingw on Windows build errors

### ISSUE THREAD

**Issue by bagder** [curl/curl#10291]
mingw now fails to build current git master on windows, see these recent [build logs](https://ci.appveyor.com/project/curlorg/curl/builds/45926363)

Pretty sure db91dbbf2c0a853 caused this regression.

**Comment by icing** [comment]
Hmm, a gcc that reports __STDC_VERSION__ as c99 and yet refuses %z. well done.

**Comment by bagder** [comment]
I don't think it hurts us too much if we just disable this check for mingw builds.
